manifests/mod/cache_disk.pp

```diff
@@ -0,0 +1,121 @@
+#   Installs and configures `mod_cache_disk`.
+#
+# @description
+#   This will install an configure the proper module depending on the used apache version, so
+#   - mod_cache_disk for apache version >= 2.4
+#   - mod_disk_cache for older apache versions
+# 
+# @param cache_root
+#   Defines the name of the directory on the disk to contain cache files.
+#   Default depends on the Apache version and operating system:
+#   - Debian: /var/cache/apache2/mod_cache_disk
+#   - FreeBSD: /var/cache/mod_cache_disk
+#   - Red Hat, Apache 2.4: /var/cache/httpd/proxy
+#   - Red Hat, Apache 2.2: /var/cache/mod_proxy
+#
+# @param cache_enable
+#   Defines an array of directories to cache, the default is none
+#
+# @param cache_ignore_headers
+#   Specifies HTTP header(s) that should not be stored in the cache.
+#
+# @param cache_dir_length
+#   The number of characters in subdirectory names
+#
+# @param cache_dir_levels
+#   The number of levels of subdirectories in the cache.
+#
+# @param cache_default_expire
+#   The default duration to cache a document when no expiry date is specified.
+#
+# @param cache_max_expire
+#   The maximum time in seconds to cache a document
+#
+# @param cache_ignore_no_lastmod
+#   Ignore the fact that a response has no Last Modified header.
+#
+# @param cache_header
+#   Add an X-Cache header to the response.
+#
+# @param cache_lock
+#   Enable the thundering herd lock.
+#
+# @param cache_ignore_cache_control
+#   Ignore request to not serve cached content to client
+#
+# @param cache_max_filesize
+#   The maximum size (in bytes) of a document to be placed in the cache
+#
+# @note
+#   Apache 2.2, mod_disk_cache installed. On Apache 2.4, mod_cache_disk installed.
+#
+# @see https://httpd.apache.org/docs/2.2/mod/mod_disk_cache.html for additional documentation on version 2.2.
+#
+# @see https://httpd.apache.org/docs/2.4/mod/mod_cache_disk.html for additional documentation on version 2.4.
+#
+class apache::mod::cache_disk (
+  $cache_root                                             = undef,
+  Array[String] $cache_enable                             = [],
+  Optional[String] $cache_ignore_headers                  = undef,
+  Optional[Integer] $cache_dir_length                     = undef,
+  Optional[Integer] $cache_dir_levels                     = undef,
+  Optional[Integer] $cache_default_expire                 = undef,
+  Optional[Integer] $cache_max_expire                     = undef,
+  Optional[Enum['Off', 'On']] $cache_ignore_no_lastmod    = undef,
+  Optional[Enum['off', 'on']] $cache_header               = undef,
+  Optional[Enum['off', 'on']] $cache_lock                 = undef,
+  Optional[Enum['Off', 'On']] $cache_ignore_cache_control = undef,
+  Optional[Integer] $cache_max_filesize                   = undef,
+) {
+  include apache
+  if $cache_root {
+    $_cache_root = $cache_root
+  }
+  elsif versioncmp($apache::apache_version, '2.4') >= 0 {
+    $_module_name = 'cache_disk'
+    $_cache_root = $::osfamily ? {
```

Comment:
Please use `$facts['os']['family']`

Reply:
Oh, you're quick. 😆  The PR is in draft. I'm currently optimizing and adding tests. Thanks!

Reply:
I saw that and didn't look too closely at details, just wanted to point out some general things that jumped out and can affect writing tests.